drakrun/drakrun/lib/postprocessing/capa_plugin/capa_processor.py

```diff
@@ -0,0 +1,428 @@
+    return capa.features.extractors.drakvuf.extractor.DrakvufExtractor.from_report(
+        calls
+    )
+
+
+def decode_json_lines(fd: BinaryIO) -> Iterator[Dict]:
+    # read and decode json lines into a list of dictionaries
+    for line in fd:
+        try:
+            # we use orjson for a small performance improvement
+            line_s = line.strip()
+            obj = orjson.loads(line_s)
+            yield obj
+        except (orjson.JSONDecodeError):
+            # sometimes Drakvuf reports bad method names and/or malformed JSON
+            logger.debug("bad drakvuf log line: %s", line)
+
+
+def dynamic_capa_analysis(
+    analysis_dir: Path,
+    rules: capa.rules.RuleSet,
+    malware_pids: Optional[List[int]] = None,
+) -> Tuple[Path, capa.render.result_document.MatchResults]:
+
+    # save all api calls and native calls into one list which gets sorted by capa later on.
+    calls = []
+
+    # read api calls
+    if (analysis_dir / "apimon.log").exists():
+        with (analysis_dir / "apimon.log").open("r", errors="ignore") as apimon_fd:
+            calls += list(decode_json_lines(apimon_fd))
+    else:
+        logger.debug("missing apimon.log file")
+
+    # read syscalls
+    if (analysis_dir / "syscall.log").exists():
+        with (analysis_dir / "syscall.log").open("r", errors="ignore") as syscall_fd:
+            calls += list(decode_json_lines(syscall_fd))
+    else:
+        logger.debug("missing syscall.log file")
+
+    # initialize the Drakvuf capa feature extractor with the captured calls
+    extractor = get_drakvuf_feature_extractor(calls)
+
+    # apply a filter to the extractor if we wish to only analyze a specific set of processes
+    if malware_pids:
+        extractor = capa.features.extractors.base_extractor.ProcessFilter(
+            extractor, malware_pids
+        )
+
+    # extract dymamic capabilities
+    capabilities, _ = capa.capabilities.common.find_capabilities(rules, extractor)
+
+    # analysis_dir is returned for conformity reasons with the ttp generation function (same routine for both static and dynamic capability extraction)
+    return analysis_dir, capabilities
+
+
+def get_process_memory_dumps(analysis_dir: Path, pid: int) -> Iterator[str]:
+    # get the memory dumps from a specific process.
+    # this is used mainly to get the memdumps by a malware process
+    with (analysis_dir / "memdump.log").open("r") as f:
+        for line in f:
+            dump = orjson.loads(line)
+            if dump["PID"] == pid:
+                yield dump["DumpFilename"]
+
+
+def static_capa_analysis(
+    dump_path: Path, rules: capa.rules.RuleSet
+) -> Tuple[Path, capa.render.result_document.MatchResults]:
+
+    # get the input file's capa format
+    try:
+        input_format = capa.helpers.get_auto_format(dump_path)
+    except capa.main.UnsupportedFormatError:
+        logger.debug("dump %s has an unsupported format", dump_path)
+        return
```

Comment:
This doesn't match the method return type. Probably should change return type to Optional[...] in this case (and make sure that callers expect it).